go-controller/pkg/node/node.go

```diff
@@ -286,54 +292,47 @@ func (n *OvnNode) WatchEndpoints() {
 		UpdateFunc: func(old, new interface{}) {
 			epNew := new.(*kapi.Endpoints)
 			epOld := old.(*kapi.Endpoints)
+			// return if there are no endpoint changes
 			if apiequality.Semantic.DeepEqual(epNew.Subsets, epOld.Subsets) {
 				return
 			}
-			newEpAddressMap := buildEndpointAddressMap(epNew.Subsets)
-			for item := range buildEndpointAddressMap(epOld.Subsets) {
-				if _, ok := newEpAddressMap[item]; !ok {
-					err := deleteConntrack(item.ip, item.port, item.protocol)
-					if err != nil {
-						klog.Errorf("Failed to delete conntrack entry for %s: %v", item.ip, err)
-					}
-				}
+			// get the service that belongs to the endpoint
+			svc, err := n.watchFactory.GetService(epNew.Namespace, epNew.Name)
```

Comment:
Can't we change this to a service watcher instead? Endpoints are not needed anymore it seems.

Reply:
this is no valid, because each node can be the Vitrual IP of the node port service, and this must run in each node that can have a client to that VIP